src/utils/cache/keypair.ts

```diff
@@ -1,20 +1,39 @@
 import type { Keypairs } from "@prisma/client";
-import { getKeypairByPublicKey } from "../../db/keypair/get";
+import { getKeypairByHash, getKeypairByPublicKey } from "../../db/keypair/get";
 
 // Cache a public key to the Keypair object, or null if not found.
 export const keypairCache = new Map<string, Keypairs | null>();
 
-export const getKeypair = async ({
-  publicKey,
-}: {
-  publicKey: string;
+/**
+ * Get a keypair by public key or hash.
+ */
+export const getKeypair = async (args: {
```

Comment:
nit: this could be made typesafe (I know we do runtime checks but still)

`type GetKeypairArgs = { publicKey: string } | {publicKeyHash: string}`

I understand current version is more convenient because it reduces upstream changes.